src/legacy/ui/public/vis/editors/default/components/default_editor_agg_group_helper.test.ts

```diff
@@ -0,0 +1,139 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AggConfig } from '../../../agg_config';
+import {
+  isAggRemovable,
+  calcAggIsTooLow,
+  isInvalidAggsTouched,
+} from './default_editor_agg_group_helper';
+import { AggsState } from './default_editor_agg_group_state';
+
+describe('DefaultEditorGroup helpers', () => {
+  let group: AggConfig;
+
+  beforeEach(() => {
+    group = [
+      {
+        id: 1,
+        title: 'Test1',
+        params: {
+          field: {
+            type: 'number',
+          },
+        },
+        group: 'metrics',
+        schema: { name: 'metric', min: 1, mustBeFirst: true },
+      },
+      {
+        id: 2,
+        title: 'Test2',
+        params: {
+          field: {
+            type: 'string',
+          },
+        },
+        group: 'metrics',
+        schema: { name: 'metric', min: 2 },
+      },
+    ];
+  });
+  describe('isAggRemovable', () => {
+    it('should return true when the number of aggs with the same schema is above the min', () => {
+      const isRemovable = isAggRemovable(group[0], group);
+
+      expect(isRemovable).toBeTruthy();
+    });
+
+    it('should return false when the number of aggs with the same schema is not above the min', () => {
+      const isRemovable = isAggRemovable(group[1], group);
+
+      expect(isRemovable).toBeFalsy();
+    });
+  });
+
+  describe('calcAggIsTooLow', () => {
+    it('should return false when agg.schema.mustBeFirst has falsy value', () => {
+      const isRemovable = calcAggIsTooLow(group[1], 0, group);
+
+      expect(isRemovable).toBeFalsy();
+    });
+
+    it('should return false when there is no different schema', () => {
+      group[1].schema = group[0].schema;
+      const isRemovable = calcAggIsTooLow(group[0], 0, group);
+
+      expect(isRemovable).toBeFalsy();
+    });
+
+    it('should return false when different schema is not less than agg index', () => {
+      const isRemovable = calcAggIsTooLow(group[0], 0, group);
+
+      expect(isRemovable).toBeFalsy();
+    });
+
+    it('should return true when agg index is greater than different schema index', () => {
+      const isRemovable = calcAggIsTooLow(group[0], 2, group);
+
+      expect(isRemovable).toBeTruthy();
+    });
+  });
+
+  describe('isInvalidAggsTouched', () => {
+    let aggsState: AggsState;
+
+    beforeEach(() => {
+      aggsState = {
+        1: {
+          valid: true,
+          touched: false,
+        },
+        2: {
+          valid: true,
+          touched: false,
+        },
+        3: {
+          valid: true,
+          touched: false,
+        },
+      };
+    });
+
+    it('should return false when there are no invalid aggs', () => {
+      const isRemovable = isInvalidAggsTouched(aggsState);
```

Comment:
The helper function is named `isInvalidAggsTouched` and is used in `agg_group` component as `isAllAggsTouched`.
Why in tests it is named as `isRemovable` ?

Reply:
This is a typo. Thanks.
Fixed.